tests/components/viam/test_config_flow.py

```diff
@@ -0,0 +1,227 @@
+"""Test the viam config flow."""
+import asyncio
+from dataclasses import dataclass
+from unittest.mock import AsyncMock, patch
+
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.viam.config_flow import CannotConnect, InvalidAuth
+from homeassistant.components.viam.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+@dataclass
+class MockLocation:
+    """Fake location for testing."""
+
+    id: int = 13
+    name: str = "home"
+
+
+@dataclass
+class MockRobot:
+    """Fake robot for testing."""
+
+    id: int = 1234
+    name: str = "test"
+
+
+def async_return(result):
+    """Allow async return value with MagicMock."""
+
+    future = asyncio.Future()
+    future.set_result(result)
+    return future
+
+
+async def test_user_form(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test that the form is served with no input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+    assert result["step_id"] == "user"
+
+    with patch(
+        "homeassistant.components.viam.config_flow.ViamHub",
+    ) as MockHub:
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "credential_type": "api-key",
```

Comment:
Also, use the constants in the tests too. Using constants and classes over string reduces typos and is easier to review and maintain the code